app/uk/gov/hmrc/slacknotifications/services/NotificationService.scala

```diff
@@ -51,19 +52,26 @@ class NotificationService @Inject()(
     notificationRequest.channelLookup match {
 
       case ChannelLookup.GithubRepository(name) =>
-        withExistingRepository(name) { repositoryDetails =>
-          withTeamsResponsibleForRepo(name, repositoryDetails) { teamNames =>
-            traverseFuturesSequentially(teamNames) { teamName =>
-              withExistingSlackChannel(teamName) { slackChannel =>
-                sendSlackMessage(fromNotification(notificationRequest, slackChannel), service, Some(teamName))
-              }
-            }.map(flatten)
-          }
-        }
+        {
+          for {
+            repositoryDetails         <- EitherT(withExistingRepository(name))
+            allTeams                  <- EitherT(withTeamsResponsibleForRepo(name, repositoryDetails))
+            (excluded, toBeProcessed)  = allTeams.partition(slackNotificationConfig.notRealTeams.contains)
+            notificationResult        <- EitherT.liftF[Future,NotificationResult, NotificationResult](toBeProcessed.foldLeftM(Seq.empty[NotificationResult]){(acc, teamName) =>
+                for {
+                  slackChannel    <- withExistingSlackChannel(teamName)
+                  notificationRes <- sendSlackMessage(fromNotification(notificationRequest, slackChannel), service, Some(teamName))
+                } yield acc :+ notificationRes
+              }.map(flatten))
+            resWithExclusions          = notificationResult.addExclusion(excluded.map(NotARealTeam.apply): _*)
+          } yield resWithExclusions
+        }.merge
 
       case ChannelLookup.SlackChannel(slackChannels) =>
-        traverseFuturesSequentially(slackChannels.toList) { slackChannel =>
-          sendSlackMessage(fromNotification(notificationRequest, slackChannel), service)
+        slackChannels.toList.foldLeftM(Seq.empty[NotificationResult]){(acc, slackChannel) =>
+          for {
```

Comment:
since just a single entry in the for-comprehension, it might be cleaner to do `sendSlackMessage(fromNotification(notificationRequest, slackChannel), service).map(acc :+ _)`

Reply:
Agreed - have changed :)